app/controllers/application_controller.rb

```diff
@@ -1,10 +1,133 @@
 require './config/environment'
 
+# ApplicationController handles requests for homepage, signup and logins
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+
+    enable :sessions
+    set :session_secret, "secret"
+  end
+
+  get '/' do
+    redirect "/tweets" if Helpers.is_logged_in?(session)
+    erb :index
+  end
+
+  get '/signup' do
+    redirect "tweets" if Helpers.is_logged_in?(session)
+    erb :"users/signup"
+  end
+
+  post '/signup' do
+    if params[:username] == "" || params[:email] == "" || params[:password] == ""
+      redirect "/signup", locals: {message: "Please submit enter all the information to complete signup."}
+    else
+      @user = User.create(
+        username: params[:username],
+        email: params[:email],
+        password: params[:password]
+      )
+      session[:user_id] = @user.id
+      redirect "/tweets", locals: {message: "Welcome to Fwitter. Thank you for signing up."}
```

Comment:
Okay I see the issue. I should change it from redirect to erb :"tweets/index". Thanks for telling me. That's weird, it just redirected me to tweets when I did it.



> On Feb 28, 2016, at 10:36 PM, Randall Reed <notifications@github.com> wrote:
> 
> In app/controllers/application_controller.rb:
> 
> > +  get '/signup' do
> > +    redirect "tweets" if Helpers.is_logged_in?(session)
> > +    erb :"users/signup"
> > +  end
> > +
> > +  post '/signup' do
> > +    if params[:username] == "" || params[:email] == "" || params[:password] == ""
> > +      redirect "/signup", locals: {message: "Please submit enter all the information to complete signup."}
> > +    else
> > +      @user = User.create(
> > +        username: params[:username],
> > +        email: params[:email],
> > +        password: params[:password]
> > +      )
> > +      session[:user_id] = @user.id
> > +      redirect "/tweets", locals: {message: "Welcome to Fwitter. Thank you for signing up."}
> 127.0.0.1 - - [28/Feb/2016:22:34:33 -0500] "POST /signup HTTP/1.1" 303 - 0.0747
> D, [2016-02-28T22:34:33.306229 #26586] DEBUG -- : (0.1ms) begin transaction
> D, [2016-02-28T22:34:33.307061 #26586] DEBUG -- : SQL (0.2ms) INSERT INTO "users" ("username", "email", "password_digest") VALUES (?, ?, ?) [["username", "rreed"], ["email", "rreed@test.com"], ["password_digest", "$2a$10$gqjg1wfIPudGIsLcv05DXekR0PLLTcyOP3a.J0ipcM1Nkl4/Z6/nu"]]
> D, [2016-02-28T22:34:33.309637 #26586] DEBUG -- : (2.4ms) commit transaction
> Unexpected error while processing request: Body yielded non-string value [:locals, {:message=>"Welcome to Fwitter. Thank you for signing up."}]
> 
> —
> Reply to this email directly or view it on GitHub.
>